buffer/src/main/java/io/netty/buffer/PoolChunk.java

```diff
@@ -130,6 +133,9 @@
  *
  */
 final class PoolChunk<T> implements PoolChunkMetric {
+    @SuppressWarnings("unchecked")
+    private static final AtomicIntegerFieldUpdater<PoolChunk<?>> PINNED_UPDATER = (AtomicIntegerFieldUpdater<PoolChunk<?>>)
+            (AtomicIntegerFieldUpdater<?>) newUpdater(PoolChunk.class, "pinnedBytes");
```

Comment:
nit: Not strong about this but we could also just use `AtomicInteger` directly here as I wouldn't expect a massive amount of these instances. That said I am not strong on it at all.

Reply:
I was mostly thinking about the number of memory indirections, tbh.

Reply:
I guess it not matters too much... That said I am not strong here like I said. So just keep it.

Reply:
Changed it anyway.